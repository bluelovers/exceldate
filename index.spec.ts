// @ts-nocheck

import exceldate from '.';

describe('exceldate', () =>
{
	it('should return the correct date', () =>
	{
		expect(exceldate(42510).toISOString()).toEqual('2016-05-20T00:00:00.000Z')
		expect(exceldate('42510').toISOString()).toEqual('2016-05-20T00:00:00.000Z')
		expect(exceldate(1).toISOString()).toEqual('1899-12-31T00:00:00.000Z')
		expect(exceldate('1').toISOString()).toEqual('1899-12-31T00:00:00.000Z')
		expect(exceldate(2).toISOString()).toEqual('1900-01-01T00:00:00.000Z')
		expect(exceldate('2').toISOString()).toEqual('1900-01-01T00:00:00.000Z')
		expect(exceldate(4242.4242).toISOString()).toEqual('1911-08-12T10:10:50.880Z')
		expect(exceldate('4242.4242').toISOString()).toEqual('1911-08-12T10:10:50.880Z')
		expect(exceldate(42738.22626859954).toISOString()).toEqual('2017-01-03T05:25:49.607Z')
		expect(exceldate('42738.22626859954').toISOString()).toEqual('2017-01-03T05:25:49.607Z')
	})

	it('should return the correct date via callbacks', () =>
	{
		exceldate(42510, (err, res) =>
		{
			expect(res.toISOString()).toEqual('2016-05-20T00:00:00.000Z')
		})
		exceldate('42510', (err, res) =>
		{
			expect(res.toISOString()).toEqual('2016-05-20T00:00:00.000Z')
		})
		exceldate(1, (err, res) =>
		{
			expect(res.toISOString()).toEqual('1899-12-31T00:00:00.000Z')
		})
		exceldate('1', (err, res) =>
		{
			expect(res.toISOString()).toEqual('1899-12-31T00:00:00.000Z')
		})
		exceldate(2, (err, res) =>
		{
			expect(res.toISOString()).toEqual('1900-01-01T00:00:00.000Z')
		})
		exceldate('2', (err, res) =>
		{
			expect(res.toISOString()).toEqual('1900-01-01T00:00:00.000Z')
		})
		exceldate(4242.4242, (err, res) =>
		{
			expect(res.toISOString()).toEqual('1911-08-12T10:10:50.880Z')
		})
		exceldate('4242.4242', (err, res) =>
		{
			expect(res.toISOString()).toEqual('1911-08-12T10:10:50.880Z')
		})
		exceldate(42738.22626859954, (err, res) =>
		{
			expect(res.toISOString()).toEqual('2017-01-03T05:25:49.607Z')
		})
		exceldate('42738.22626859954', (err, res) =>
		{
			expect(res.toISOString()).toEqual('2017-01-03T05:25:49.607Z')
		})

		expect.assertions(10);
	})

	it('should throw an error for invalid inputs', () =>
	{
		expect(() => exceldate('foo')).toThrowErrorMatchingSnapshot()
		expect(() => exceldate(false)).toThrowErrorMatchingSnapshot()
		expect(() => exceldate()).toThrowErrorMatchingSnapshot()
	})

	it('should return errors for invalid inputs via callback', () =>
	{
		exceldate('foo', (err, res) =>
		{
			expect(res).toBeUndefined()
			expect(err.toString()).toEqual('Error: First argument could not be parsed.')
		})

		exceldate(null, (err, res) =>
		{
			expect(res).toBeUndefined()
			expect(err.toString()).toEqual('Error: No first argument provided, nothing to convert.')
		})

		expect.assertions(4);
	})
})
