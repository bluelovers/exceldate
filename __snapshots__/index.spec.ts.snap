// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exceldate should throw an error for invalid inputs 1`] = `"First argument could not be parsed."`;

exports[`exceldate should throw an error for invalid inputs 2`] = `"No first argument provided, nothing to convert."`;

exports[`exceldate should throw an error for invalid inputs 3`] = `"No first argument provided, nothing to convert."`;
